### PR TITLE
52811

### PR DESCRIPTION
**Pull request by Yatekii**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ Every nRF chip has its own crate, listed below:
 |-------|------|-----------|
 | [`nrf51-hal`](./nrf51-hal) | [![docs.rs](https://docs.rs/nrf51-hal/badge.svg)](https://docs.rs/nrf51-hal) | [![crates.io](https://img.shields.io/crates/d/nrf51-hal.svg)](https://crates.io/crates/nrf51-hal) |
 | [`nrf52810-hal`](./nrf52810-hal) | [![docs.rs](https://docs.rs/nrf52810-hal/badge.svg)](https://docs.rs/nrf52810-hal) | [![crates.io](https://img.shields.io/crates/d/nrf52810-hal.svg)](https://crates.io/crates/nrf52810-hal) |
+| [`nrf52810-hal`](./nrf52811-hal) | [![docs.rs](https://docs.rs/nrf52811-hal/badge.svg)](https://docs.rs/nrf52811-hal) | [![crates.io](https://img.shields.io/crates/d/nrf52811-hal.svg)](https://crates.io/crates/nrf52811-hal) |
```
Comment (Dirbaio) [inline README.md:21]: nrf52810 -> nrf52811

Comment (Yatekii) [reply README.md:21]: Ofc I forgot one m( Thanks!

File: nrf-hal-common/src/comp.rs [hunk 2]
```diff
@@ -302,10 +302,10 @@ comp_input_pins! {
     P0_28<Input<Floating>> => PSEL_A::ANALOGINPUT4,
     P0_29<Input<Floating>> => PSEL_A::ANALOGINPUT5,
     P0_30<Input<Floating>> => PSEL_A::ANALOGINPUT6,
-    P0_31<Input<Floating>> => PSEL_A::ANALOGINPUT7,
+    P0_31<Input<Floating>> => PSEL_A::ANALOGINPUT6,
```
Comment (jacobrosenthal) [inline nrf-hal-common/src/comp.rs:305]: Should still be ain7 right?

Comment (Yatekii) [reply nrf-hal-common/src/comp.rs:305]: dang, good spot, thanks!

**Comment by jacobrosenthal** [comment]
This needs to get removed too? Or do we intend to define this environment variable
https://github.com/nrf-rs/nrf-hal/blob/3ce300c0d4c3ed0d62bbf247ce14cb2de9f0bade/nrf-hal-common/build.rs

**Comment by Yatekii** [comment]
> This needs to get removed too? Or do we intend to define this environment variable
> https://github.com/nrf-rs/nrf-hal/blob/3ce300c0d4c3ed0d62bbf247ce14cb2de9f0bade/nrf-hal-common/build.rs

I don't see why this should be removed? It is helpful? :)

**Comment by jacobrosenthal** [comment]
ah yeah nevermind :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,7 @@
 - PWM module ([#200]).
 - I2S module ([#201]).
 - SPIS module ([#226]).
+- Add support for the nRF52811.
```
Comment (jonas-schievink) [inline CHANGELOG.md:15]: PR link is missing (though I'm not sure we want to keep adding them)

File: nrf52811-hal/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+default-features = false
+features = ["52811"]
+version = "=0.11.1"
+
+[dependencies.embedded-hal]
+features = ["unproven"]
+version = "0.2.3"
+
+[features]
+doc = []
```
Comment (jonas-schievink) [inline nrf52811-hal/Cargo.toml:32]: This seems unused

File: xtask/src/lib.rs [hunk 5]
```diff
@@ -34,6 +41,7 @@ pub fn feature_to_target(feat: &str) -> &str {
     match feat {
         "51" => "thumbv6m-none-eabi",
         "52810" => "thumbv7em-none-eabi",
+        "52811" => "thumbv7em-none-eabi",
```
Comment (jonas-schievink) [inline xtask/src/lib.rs:44]:
```suggestion
        "52810" |
        "52811" => "thumbv7em-none-eabi",
```

**Review by jonas-schievink — COMMENTED** [review]
Looks good! Needs a rebase to fix the merge conflicts though.

Have the problems with the SVD been fixed or are there still workarounds needed?

File: nrf52811-hal/Cargo.toml (hunk 4 above)
Comment (Yatekii) [reply nrf52811-hal/Cargo.toml:32]: Do you mean the embedded-hal? Not sure whether GH is broken or I missunderstand because of other reasons ;)

**Comment by Yatekii** [comment]
I rebased this onto master to make it build again. Ofc it got into conflict with the PR I merged earlier today =)

I patched the SVD manually to be in line with all the other pac crates. It was my understanding from the comments on Matrix that all the SVDs have manual adjustments. It only really was changes in the SPI section (swapoped SPI0 and SPI1 naming wise :))

File: nrf52811-hal/Cargo.toml (hunk 4 above)
Comment (jonas-schievink) [reply nrf52811-hal/Cargo.toml:32]: I meant that the `doc` feature seems unused

Comment (Yatekii) [reply nrf52811-hal/Cargo.toml:32]: Ah, I see :) I guess it's unused in the 810 crate then too :)

**Comment by jonas-schievink** [comment]
> I patched the SVD manually to be in line with all the other pac crates. It was my understanding from the comments on Matrix that all the SVDs have manual adjustments. It only really was changes in the SPI section (swapoped SPI0 and SPI1 naming wise :))

Yup, that's good.

**Comment by Yatekii** [comment]
Ok, I think all the changes have been made as requested :)

**Pull request merged**